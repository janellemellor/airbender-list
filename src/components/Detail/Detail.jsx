import React from 'react';
import PropTypes from 'prop-types';

const Detail = ({ image, name, affiliation, allies, enemies }) => (
  <>
    <img src={image} alt={name} />
    <h3>{name}</h3>
    <p>{affiliation}</p>
    <p>Allies:{allies}</p>
    <p>Enemies:{enemies}</p>
  </>
);

Detail.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  allies: PropTypes.array,
  enemies: PropTypes.array,
};

export default Detail;
