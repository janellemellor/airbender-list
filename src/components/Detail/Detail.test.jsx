import React from 'react';
import { shallow } from 'enzyme';
import Detail from '../Detail/Detail.jsx';

describe('Detail component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Detail image='https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441' name='46th Earth King' affiliation='Earth Kingdom Earth Kingdom Royal Family' allies='Royal Earthbender Guards' enemies='Chin' />);
    expect(wrapper).toMatchSnapshot();
  });
});

