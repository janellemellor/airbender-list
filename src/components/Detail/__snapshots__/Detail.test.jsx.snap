// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component matches a snapshot 1`] = `
<Fragment>
  <img
    alt="46th Earth King"
    src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
  />
  <h3>
    46th Earth King
  </h3>
  <p>
    Earth Kingdom Earth Kingdom Royal Family
  </p>
  <p>
    Allies:
    Royal Earthbender Guards
  </p>
  <p>
    Enemies:
    Chin
  </p>
</Fragment>
`;
